rest-server/src/controllers/job.js

```diff
@@ -120,23 +86,19 @@ const update = (req, res) => {
     if (err) {
       logger.warn('update job %s error\n%s', name, err.stack);
       if (err.message === 'VirtualClusterNotFound') {
-        return res.status(500).json({
-          error: 'JobUpdateWithInvalidVirtualCluster',
-          message: `job update error: could not find virtual cluster ${data.virtualCluster}`,
-        });
+        err.status = httpStatus.INTERNAL_SERVER_ERROR;
+        err.message = `job update error: could not find virtual cluster ${data.virtualCluster}`;
+        next(err);
       } else if (err.message === 'NoRightAccessVirtualCluster') {
-        return res.status(401).json({
-          error: 'JobUpdateWithNoRightVirtualCluster',
-          message: `job update error: no virtual cluster right to access ${data.virtualCluster}`,
-        });
+        err.status = httpStatus.UNAUTHORIZED;
+        err.message = `job update error: no virtual cluster right to access ${data.virtualCluster}`;
+        next(err);
       } else {
-        return res.status(500).json({
-          error: 'JobUpdateError',
-          message: err.message,
-        });
+        err.status = httpStatus.INTERNAL_SERVER_ERROR;
```

Comment:
Showing ‘JobUpdateError’ only usually makes users frustrated and confused (we have many reports about this). Suggest include more detailed error description.

Reply:
"JobUpdateError" has been removed already

Reply:
still, is it possible to include more detailed err msg here?

---
In reply to: [193630715](https://github.com/Microsoft/pai/pull/660#discussion_r193630715) [](ancestors = 193630715)